frontend/app/module/NotFoundProvider.tsx

```diff
@@ -0,0 +1,29 @@
+import { createContext, useMemo, useState } from "react";
+
+export interface iNotFoundContext {
+  notFound: boolean;
+  setNotFound: (notFound: boolean) => void;
+}
+
+export const NotFoundContext = createContext<iNotFoundContext>({
+  notFound: false,
+  setNotFound: (notFound: boolean) => {
+    notFound;
+  },
+});
+
+export const NotFoundProvider = (props: object) => {
```

Comment:
My IDE can't find any usages of the `NotFoundProvider`.

Reply:
Funny, I would expect the linter to catch this. Since we use the `errorElement` logic of the router, we don't need a provider or context at all. Thanks!

Reply:
Looks like we need [`eslint-plugin-import`](https://www.npmjs.com/package/eslint-plugin-import) for that, because `eslint` only does unused vars and unused function params.